barretenberg/cpp/src/barretenberg/commitment_schemes/pcs_test_instance_witness_generator.hpp

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include "barretenberg/ecc/curves/bn254/bn254.hpp"
+#include "barretenberg/polynomials/polynomial.hpp"
+#include "barretenberg/transcript/transcript.hpp"
+#include "commitment_key.test.hpp"
+
+namespace bb {
+/**
+ * @brief Constructs random polynomials, computes commitments and corresponding evaluations.
+ *
+ * @tparam Curve
+ */
+template <typename Curve> struct PCSInstanceWitnessGenerator {
+  public:
+    using CommitmentKey = bb::CommitmentKey<Curve>;
+    using Fr = typename Curve::ScalarField;
+    using Commitment = typename Curve::AffineElement;
+    using Polynomial = bb::Polynomial<Fr>;
+
+    std::shared_ptr<CommitmentKey> ck;
+    std::vector<Polynomial> polynomials;
```

Comment:
I would name this unshifted_polynomials and the ones below to_be_shifted_polynomials for clarity